hikari/impl/rest.py

```diff
@@ -2425,6 +2425,42 @@ async def edit_welcome_screen(
         response = typing.cast(data_binding.JSONObject, raw_response)
         return self._entity_factory.deserialize_welcome_screen(response)
 
+    async def fetch_guild_membership_gate(
+        self, guild: snowflakes.SnowflakeishOr[guilds.PartialGuild]
+    ) -> guilds.MembershipGate:
+        route = routes.GET_GUILD_MEMBER_VERIFICATION.compile(guild=guild)
+        response = await self._request(route)
+
+        # If the member screen of a guild has never been set
+        # then Discord will respond with a 204 (no content).
+        if not response:
+            return guilds.MembershipGate(description=None, fields=[], updated_at=None)
```

Comment:
Should this maybe return a `None` since there's no member screen? ... Unless I'm misinterpreting the comment, that is.

Reply:
We'd never return a None in REST for a GET request cause that's what NotFoundError but even then for this endpoint that only applies to if the guild existed or not. Regardless of whether this is enabled or not as long as the guild exists this entity should also exist just Discord rather than having a special case which handles when it's never been set and makes sure the returned data fits their defined model for the endpoint (as opposed to idk just returning the empty string they fetched from a database) they instead seem to just return nothing which is undocumented behaviour and inconsistent.